cmd/influxd/upgrade/upgrade.go

```diff
@@ -478,54 +484,85 @@ func runUpgradeE(cmd *cobra.Command, options *options, verbose bool) error {
-	if sourceOpts.dbDir != "" {
-		fi, err := os.Stat(sourceOpts.dbDir)
+// validatePaths ensures that all paths pointing to V1 inputs are usable by the upgrade command.
+func (o *optionsV1) validatePaths() error {
+	if o.dbDir != "" {
+		fi, err := os.Stat(o.dbDir)
 		if err != nil {
-			return fmt.Errorf("1.x DB dir '%s' does not exist", sourceOpts.dbDir)
+			return fmt.Errorf("1.x DB dir '%s' does not exist", o.dbDir)
 		}
 		if !fi.IsDir() {
-			return fmt.Errorf("1.x DB dir '%s' is not a directory", sourceOpts.dbDir)
+			return fmt.Errorf("1.x DB dir '%s' is not a directory", o.dbDir)
 		}
 	}
 
-	metaDb := filepath.Join(sourceOpts.metaDir, "meta.db")
+	metaDb := filepath.Join(o.metaDir, "meta.db")
 	_, err := os.Stat(metaDb)
 	if err != nil {
 		return fmt.Errorf("1.x meta.db '%s' does not exist", metaDb)
 	}
 
-	if targetOpts.configPath != "" {
-		if _, err := os.Stat(targetOpts.configPath); err == nil {
-			return fmt.Errorf("file present at target path for upgraded 2.x config file '%s'", targetOpts.configPath)
+	return nil
+}
+
+// validatePaths ensures that none of the paths pointing to V2 outputs refer to existing files.
+func (o *optionsV2) validatePaths() error {
+	if o.configPath != "" {
+		if _, err := os.Stat(o.configPath); err == nil {
+			return fmt.Errorf("file present at target path for upgraded 2.x config file '%s'", o.configPath)
 		}
 	}
 
-	if _, err = os.Stat(targetOpts.boltPath); err == nil {
-		return fmt.Errorf("file present at target path for upgraded 2.x bolt DB: '%s'", targetOpts.boltPath)
+	if _, err := os.Stat(o.boltPath); err == nil {
+		return fmt.Errorf("file present at target path for upgraded 2.x bolt DB: '%s'", o.boltPath)
 	}
 
-	if fi, err := os.Stat(targetOpts.enginePath); err == nil {
+	if fi, err := os.Stat(o.enginePath); err == nil {
 		if !fi.IsDir() {
-			return fmt.Errorf("upgraded 2.x engine path '%s' is not a directory", targetOpts.enginePath)
+			return fmt.Errorf("upgraded 2.x engine path '%s' is not a directory", o.enginePath)
 		}
-		entries, err := ioutil.ReadDir(targetOpts.enginePath)
+		entries, err := ioutil.ReadDir(o.enginePath)
 		if err != nil {
 			return err
 		}
 		if len(entries) > 0 {
-			return fmt.Errorf("upgraded 2.x engine directory '%s' must be empty", targetOpts.enginePath)
+			return fmt.Errorf("upgraded 2.x engine directory '%s' must be empty", o.enginePath)
 		}
 	}
 
-	if _, err = os.Stat(targetOpts.cliConfigsPath); err == nil {
-		return fmt.Errorf("file present at target path for 2.x CLI configs '%s'", targetOpts.cliConfigsPath)
+	if _, err := os.Stat(o.cliConfigsPath); err == nil {
+		return fmt.Errorf("file present at target path for 2.x CLI configs '%s'", o.cliConfigsPath)
+	}
+
+	if _, err := os.Stat(o.cqPath); err == nil {
+		return fmt.Errorf("file present at target path for exported continuous queries '%s'", o.cqPath)
+	}
+
+	return nil
+}
+
+// clearPaths deletes any files already present at the specified V2 output paths.
+func (o *optionsV2) clearPaths() error {
```

Comment:
Seems like bad UX to have `--force` with default arguments remove all the user's data. Maybe a new arg for `--remove-existing-v2` and a helpful error message if the user runs into this situation (`Existing v2 data detected at <blah>, use --remove-existing-v2 to force migration`)